pkg/internal/approver/manager/predicate/predicate.go

```diff
@@ -77,6 +86,74 @@ func SelectorIssuerRef(_ context.Context, cr *cmapi.CertificateRequest, policies
 	return matchingPolicies, nil
 }
 
+// SelectorNamespace is a Predicate that returns the subset of given policies
+// that have an `spec.selector.namespace` matching the `metadata.namespace` of
+// the request. SelectorNamespace will match with `namespace.matchNames` on
+// namespaces using wilcards "*". Empty selector is equivalent to "*" and will
+// match on any Namespace.
+func SelectorNamespace(lister client.Reader) Predicate {
+	return func(ctx context.Context, request *cmapi.CertificateRequest, policies []policyapi.CertificateRequestPolicy) ([]policyapi.CertificateRequestPolicy, error) {
+		var matchingPolicies []policyapi.CertificateRequestPolicy
+
+		var namespace corev1.Namespace
+		if err := lister.Get(ctx, client.ObjectKey{Name: request.Namespace}, &namespace); err != nil {
```

Comment:
Would it be possible to only execute this Get() call when we are certain that we have to retrieve the labels of the namespace?

Reply:
eg. if no Namespace selectors are != nil, we can skip this retrieval

Reply:
We could.. however note that we are using a lister here which is metadata caching Namespaces so should be cheap.

I've gone ahead and added lazy fetching anyway.

Reply:
PTAL